functions/index.js

```diff
@@ -1,156 +1,389 @@
+            body: JSON.stringify(data.Attributes),
+            headers: {
+              'Access-Control-Allow-Origin': '*'
+            }
+          });
+      } else {
+        callback(null, {
+          statusCode: 400,
+          body: "There was an error.",
+          headers: {
+            'Access-Control-Allow-Origin': '*'
+          }
+        });
+      }
+    })
+    .catch((err) => {
+      callback(null, {
+        statusCode: 400,
+        body: JSON.stringify({
+          message: `Error: ${err}`
+        })
+      })
+    });
   }
-}
 
-async function modifyGame(event){
-  let updateExpression = [];
-  const modifyParameters = [ 'description', 'genre', 'yearReleased', 'developer', 'console' ];
-  let expressionAttributeNames={};
-  let expressionAttributeValues = {};
-  
-  //Generate dynammic update expression based on allowed parameters
-  for (let parameter in event.queryStringParameters) {
-    if (modifyParameters.includes(parameter)) {
-      updateExpression.push(`#${parameter} = :${parameter}`);
-      expressionAttributeNames['#'+parameter] = parameter ;
-      expressionAttributeValues[':'+parameter]=event.queryStringParameters[parameter];  
+  async function deleteGame(event) {
+    await returnDeleteData(event)
+    .then((data) => {
+      if (data.hasOwnProperty('code') && data.hasOwnProperty('message') && data.hasOwnProperty('statusCode')) {
+        switch (data.code) {
+          default:
+            callback(null, {
+              statusCode: `${data.statusCode}`,
+              body: `Message: ${data.message}\nStatus Code: ${data.statusCode}`,
+              headers: {
+                'Access-Control-Allow-Origin': '*'
+              }
+            });
+            break;
+        }
+      } else if (data) {
+        //DynamoDB returns an empty JSON object when attempting to delete an item that isn't in the table
+        switch (Object.keys(data).length) {
+          case 0:
+            callback(null, {
+              statusCode: 404,
+              body: `Message: ${JSON.parse(event.body).gameName} not found\nStatus Code: 404`,
+              headers: {
+                'Access-Control-Allow-Origin': '*'
+              }
+            });
+            break;
+          default:
+            callback(null, {
+              statusCode: 200,
+              body: JSON.stringify(data.Attributes),
+              headers: {
+                'Access-Control-Allow-Origin': '*'
+              }
+            });
+        }
+      } else {
+        callback(null, {
+          statusCode: 400,
+          body: "There was an error.",
+          headers: {
+            'Access-Control-Allow-Origin': '*'
+          }
+        })
+      }
+    })
+    .catch((err) => {
+      callback(null, {
+        statusCode: 400,
+        body: JSON.stringify({
+          message: `Error: ${err}`
+        })
+      })
+    });
+  }
+
+  //DynamoDB Data Access Functions
+  async function returnGetData(event){
```

Comment:
I think you're heading in the right direction with these functions but I think you want to take them one step further. Rather than just writing functions to make your Dynamo calls for you, I'd use these as a way to model your 'Game' objects. This method would be much more clear if it were named `getGame()` for example and even better is if the parameters were more specific so that anyone else making a game would know exactly what was needed. Imagine: 

`getGame(gameName)`. 

I'd also avoid just passing the raw dynamo response back to the caller. As someone calling your getGame function I shouldn't have to know how to also parse dynamo responses or how to map them to errors. Just manage that in this function, return the most specific information and throw an exception if something goes wrong (like `throw 'game not found'`)

As for the response you might also want to think about creating a 'game' type [object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects)

That'll also help with parameters for other functions like `createGame(new Game(gameName, description...)`

(you might want to also consider moving all this into a separate module too)